File: Leetcode/Count Number of Teams/Count Number of Teams.java
// There are n soldiers standing in a line. Each soldier is assigned a unique rating value.

// You have to form a team of 3 soldiers amongst them under the following rules:

// Choose 3 soldiers with index (i, j, k) with rating (rating[i], rating[j], rating[k]).
// A team is valid if: (rating[i] < rating[j] < rating[k]) or (rating[i] > rating[j] > rating[k]) where (0 <= i < j < k < n).
// Return the number of teams you can form given the conditions. (soldiers can be part of multiple teams).


class Solution {
    public int numTeams(int[] rating) {
        int count = 0;
        // Traversing rating array to make a valid team. 
        for(int i = 0 ; i < rating.length - 2 ; i++){
            // j will start from next to i (rating next to rating of soldier i)
            for(int j = i+1; j < rating.length - 1; j++){
                // k will start from next to j (rating next to rating of soldier j)
                for(int k = j+1; k< rating.length ; k++){
                    // Condition from the question
                    if((rating[i] < rating[j] && rating[j] < rating[k]) || (rating[i] > rating[j] && rating[j] > rating[k]))
                       count++;
                }
            }
        }
                       return count;
    }
}

File: Leetcode/Count Number of Teams/Count number of teams.c

//C Program to count number of teams
int numTeams(int* rating, int ratingSize){
    int *rem;
    int ans=0;
    //allocating size for rem
    rem = calloc(sizeof(int),ratingSize);
    rem[0] = 0;
    
    //forward
    for(int i=1; i<ratingSize; i++) {
        
        for(int j=0; j < i; j++) {
            //checking the condition
            if(rating[i] > rating[j]) {
                rem[i]++;
                ans += rem[j];
            }        
        }
    }
    
    rem[ratingSize-1] = 0;
        //backward   
    for(int i=ratingSize-2; i>=0; i--) {
        rem[i] = 0;
        for(int j = ratingSize-1; j > i; j--) {
            //checking the condition
            if(rating[i] > rating[j]) {
                rem[i]++;
                ans += rem[j];
                
            } 
        }
        
    }
     //return the answer
    return ans;
}

